[logical-1985516] iP

logical-1985516 opened the pull request:
### Luke ### 

> Today is the perfect day to get epic things done.

~~Duke~~ Luke is the **best** to-do list you could ask for! It is:
- easy to use
- tidy
- fast

Features:
- [X] Manage tasks
- [ ] Search to-dos
- [ ] Personality

How to use it:
1. Go to [this page](https://github.com/logical-1985516/ip/releases/tag/v0.1) and download the .jar file
2. Double click this file
3. Run commands on it and follow the instructions on the app

If you want to play around with the code/app and test it out, just run this method in the codebase:
```
public static void main(String[] args) throws IOException, TaskNotFoundException, UnknownCommandException,
        InvalidSyntaxException, InvalidDateTimeException {
    // Details omitted
}
```

<p> Have fun 😁 </p>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

logical-1985516 committed:
Add new name, greet and exit

logical-1985516 committed:
Add echo

logical-1985516 committed:
Add List Class that supports listing and adding

logical-1985516 committed:
Add Task class

logical-1985516 committed:
Change String in tasks in List to Task

logical-1985516 committed:
Add mark as (not) done

logical-1985516 committed:
Add ToDo, Event, Deadline

logical-1985516 committed:
Change text slightly to pass test case

logical-1985516 committed:
Add exception handling and minor text changes

logical-1985516 committed:
Add delete

logical-1985516 committed:
Fix minor typo

logical-1985516 committed:
Create file/directory on first time and change List methods for savefile

logical-1985516 committed:
Create savefile/directory on first time and change List methods for file

logical-1985516 committed:
Remove redundant this keyword to meet coding standards

logical-1985516 committed:
Merge branch 'branch-Level-7'

logical-1985516 committed:
Store dueDate in Deadline as LocalDateTime; change toString()

logical-1985516 committed:
Parse date and time - throws exception if given format is wrong

logical-1985516 committed:
Fix deadline parsing and toString()

Leaving description blank in input caused the /to and date to become
the new description of deadline. toString() now allows single numbered
day in the input date.

logical-1985516 committed:
Add date parsing and update toString() for Event

logical-1985516 committed:
Merge branch 'branch-Level-8'

logical-1985516 committed:
Move Main logic into Parser and ui; expand InvalidSyntaxException

Typing "bye ..." or "list ..." would now throw InvalidSyntaxException
instead of UnknownCommandException. This makes it clearer to the user
on what to do

logical-1985516 committed:
Standardize ordering of commands; add mark/unmark to exception message

logical-1985516 committed:
Add delete logic to Parser

logical-1985516 committed:
Add Storage that stores and loads tasks from savefile

logical-1985516 committed:
Merge with branch-A-MoreOOP

logical-1985516 committed:
Merge branch 'branch-A-MoreOOP'

logical-1985516 committed:
Clean up code: import, use constant LINE for line, set isDone to private

logical-1985516 committed:
Put all classes into packages

logical-1985516 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

logical-1985516 committed:
Add Gradle support and fix style violations

logical-1985516 committed:
Move print statements from List to Ui

logical-1985516 committed:
Add JUnit tests and move printing of (un)mark from Task to Ui

logical-1985516 committed:
Add JavaDoc

logical-1985516 committed:
Add line breaks to adhere to coding standard

logical-1985516 committed:
Add "is" in front of "validTaskNum" so it sounds like boolean

logical-1985516 committed:
Add findTasks

logical-1985516 committed:
Merge branch 'branch-A-JavaDoc'

logical-1985516 committed:
Merge branch 'branch-A-CodingStandard'

logical-1985516 committed:
Merge branch 'branch-Level-9'

blaukc commented on src/main/java/kitchensink/Ui.java:
```diff
@@ -0,0 +1,74 @@
+        return input;
+    }
+
+    public void welcome() {
```
Should this be sayWelcome?

blaukc commented on src/main/java/kitchensink/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+        String minute = dueDate.getMinute() < 10 ? "0" + dueDate.getMinute()
+                : String.valueOf(dueDate.getMinute());
+        String hour = dueDate.getHour() < 10 ? "0" + dueDate.getHour()
+                : String.valueOf(dueDate.getHour());
```
I think ternary operations should be written like this (present in other places as well):
```
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;
```

blaukc reviewed (commented):
Not much issue with adhering to coding standard. Good job !!!! 🤣😎😂😀

logical-1985516 committed:
Move each ternary clause to its own line

logical-1985516 committed:
Merge branch 'branch-A-CheckStyle'

logical-1985516 committed:
Add GUI, modify parse to return String instead of printing it

logical-1985516 committed:
Fix minor bugs and clean up comments

logical-1985516 committed:
Merge branch 'branch-Level-10'

faaheem13 commented on src/main/java/kitchensink/Ui.java:
```diff
@@ -0,0 +1,74 @@
+//    }
+
+    public String welcome() {
+        return (LINE + "Hello! I'm KitchenSink!\n" + "What can I do for you?\n" + LINE);
```
Very unique naming of the chatbot, I appreciate it!

faaheem13 commented on src/main/java/kitchensink/Parser.java:
```diff
@@ -0,0 +1,150 @@
+     * @throws TaskNotFoundException    If the task is not found, as the task number is out of bounds.
+     * @throws IOException              As storage reads/writes save file.
+     * @throws InvalidSyntaxException   If the command is valid, but not in the correct format.
+     * @throws InvalidDateTimeException If the date/time given is not in the correct format.
```
extremely detailed java documentation! leaves no room for questions to be asked!

faaheem13 reviewed (commented):
LGTM! Extremely detailed and well done work!

logical-1985516 committed:
Add a few assert statements in Parser

There are few assert statements as
1. It is impossible to check for all possible inputs by the user
2. The cases are already handled by a bunch of conditionals
In the case of "bye" and "list", it is possible to exhaust all
possibilities.

logical-1985516 committed:
Return exception message to display it in GUI

logical-1985516 committed:
Throw exception if the save file is in the wrong String format

The GUI would not load at all as to prevent the user from using the app.
If the app were to be used, how should the save file be loaded? Using a
new save file would delete all the data (which is undesirable), while
finding an exception at the start of the save file would also require
deleting all the data (anything after the problem). Thus, not loading
the GUI tells the user about the problem with save file and gives them
a chance to fix the save file.

logical-1985516 committed:
Add assert statements, mostly to prevent null pointer exceptions

Parameters are checked to prevent such exceptions. clearData has a false
assertion to ensure that user data is not accidentally cleared. While
this leads to assertion errors in ListTest and StorageTest, there is
no easy way to clear the storage easily for the other tests.

logical-1985516 committed:
Merge branch 'branch-A-Assertions'

logical-1985516 committed:
Standardise method names in Ui with "display" prefix

logical-1985516 committed:
Add assertions for exceptions

logical-1985516 committed:
Merge pull request #2 from logical-1985516/branch-A-Assertions

Add assertions for exceptions

logical-1985516 committed:
Remove dead code, abstract each switch case into a method

The methods used to abstract switch cases are private as they are only
used in the class itself. This helps to keep each method short so that
it is easier to trace through them. To reduce repetition, some methods
may not be fully suitable for some cases, but are used as long as they
are similar enough.

logical-1985516 committed:
Merge pull request #3 from logical-1985516/branch-A-CodeQuality

Remove dead code, abstract each switch case into a method

logical-1985516 committed:
Fix coding standards

logical-1985516 committed:
Add help command that displays the user guide

Warning is displayed at the top to caution users so that they do not
lose their save file accidentally. Examples are given for commands
that are harder to understand (e.g. INDEX). It also tells the user
how to interpret the task list.

logical-1985516 committed:
Fix isValidTaskNum and assert statement in saveTasks

As taskNum refers to the index in ArrayList. The assert statement
now allows the change to be -1 to allow deletion of tasks.

logical-1985516 committed:
Load sample data upon first use, update displayHelp accordingly

Sample data is loaded to show how the tasks look like in the task list.
displayHelp now mentions that sample data is loaded so that the
user does not become confused by it.

logical-1985516 committed:
Merge pull request #4 from logical-1985516/branch-BCD-Extension

Branch bcd extension

logical-1985516 committed:
Change chatbot name to match that in GFMD markdown

logical-1985516 committed:
Merge pull request #5 from logical-1985516/branch-BCD-Extension

Change chatbot name to match that in GFMD markdown

logical-1985516 committed:
Fix bug where mark, unmark delete n (n is list length) throws exception

This is because exception is thrown when index is not in the range 0 to
n - 1. This is adjusted by shifting the range of values up by 1.
Changed style slightly and cleaned up testing.

logical-1985516 committed:
Update README.md

logical-1985516 committed:
Add JavaDocs to remaining methods

logical-1985516 committed:
Merge branch 'master' of https://github.com/logical-1985516/ip

logical-1985516 committed:
Update README.md

logical-1985516 committed:
Update jar file name

logical-1985516 committed:
Remove Ui LINE as Ui already separates messages and they are distracting

logical-1985516 committed:
Merge branch 'master' of https://github.com/logical-1985516/ip

logical-1985516 committed:
Add Ui.png and remove lines from README

logical-1985516 committed:
Renmae Ui.png.png to Ui.png and insert it into README

logical-1985516 committed:
Change save file name from duke to luke

logical-1985516 committed:
Add a section on explaining what happens if save file is corrupted

This can help the user figure out why the jar file is not running.

logical-1985516 committed:
Display welcome message upon load

logical-1985516 committed:
Make Task abstract, remove blank line and include help in exception

logical-1985516 committed:
Add JavaDoc to remove checkStyle violation

logical-1985516 committed:
Fix bug where invalid find syntax returns invalid todo syntax message
